System/RTETools.cpp

```diff
@@ -107,4 +107,28 @@ namespace RTE {
 	bool WithinBox(Vector &point, Vector &boxPos, float width, float height) {
 		return point.m_X >= boxPos.m_X && point.m_X < (boxPos.m_X + width) && point.m_Y >= boxPos.m_Y && point.m_Y < (boxPos.m_Y + height);
 	}
-}
\ No newline at end of file
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	std::string RoundFloatToPrecision(float input, int precision, int roundingMode) {
+		if (roundingMode == 0) {
+			std::stringstream floatStream;
+			floatStream << std::fixed << std::setprecision(precision) << input;
+			return floatStream.str();
+		} else {
+			RTEAssert(powf(10.0F, static_cast<float>(precision)) < std::numeric_limits<float>::max(), "PRECISION SET GREATER THAN EXPONENT MAX!");
+			RTEAssert(powf(10.0F, static_cast<float>(precision)) > 0, "NEGATIVE PRECISION WILL YEILD DIVIDE BY ZERO ERROR!");
+			RTEAssert(input < (std::numeric_limits<float>::max() / powf(10.0F, static_cast<float>(precision))), "VALUE WILL EXCEED NUMERIC LIMITS WITH PRECISION " + std::to_string(precision));
+			if (roundingMode == 1) {
+				float roundingBuffer = floor(input * powf(10.0F, static_cast<float>(precision))) / powf(10.0F, static_cast<float>(precision));
```

Comment:
You may be able to compress this by storing the rounding mode function (floor or ceil) in a variable or using it in a ternary. Basically, would be nice to remove some duplicated code.

Reply:
Done, if you don't like the switch-case let me know.